### ISSUE TITLE
Check if the VeleroBackup exists for unique name generated using custom UUID and then re-try creation of VelerBackup

### ISSUE THREAD

**Issue by shubham-pampattiwar** [migtools/oadp-non-admin#102]
Ref discussion: https://github.com/migtools/oadp-non-admin/pull/101#discussion_r1803508632

**Comment by shawn-hurley** [comment]
Now that we are using labels for the connection between the two, can we use generate name and have the api server deal with this?

**Comment by mateusoliveira43** [comment]
@shawn-hurley I think that is a good idea. We currently use the value of the label also as the object name (which is 63 char length, but could be 253 char length). Is there a function in the libraries we use that already implements this for us?

**Comment by kaovilai** [comment]
Per my [comments](https://github.com/migtools/oadp-non-admin/issues/90#issuecomment-2375241934) earlier, apiserver already has retry logic on generateName it creates for objects. I believe Shawn is not asking us to use any libs to implement this.. but to create object without specifying name. Just the generateName in metadata.

> Generated values
> Some values of an object are typically generated before the object is persisted. It is important not to rely upon the values of these fields set by a dry-run request, since these values will likely be different in dry-run mode from when the real request is made. Some of these fields are:
> 
> name: if generateName is set, name will have a unique random name

[src](https://kubernetes.io/docs/reference/using-api/api-concepts/#generated-values)

**Comment by mpryc** [comment]
From what is worth I think the retry logic on name is a very corner case now, so I would focus on more important issues to be addressed first.

**Comment by kaovilai** [comment]
I think this is less about discussing retry specifically, but to remove the need for UUID code to exists inside NAC. Just use name generated via generateName from apiserver.

**Comment by shubham-pampattiwar** [comment]
@kaovilai partially correct. Yes we can use GenerateName instead of using custom UUID for name. Currently custom UUID is used for establishing NAB - VB mapping and also for NAB name (but not a necessity here). GenerateName would be an improvement but we still need custom UUID for mapping purposes we already discussed. So yes we could use GenerateName for NAB name but custom UUID code still needs to reside. Hope this clears up any confusion on this.

**Comment by kaovilai** [comment]
User create NAB
- NAB creates VeleroBackup with only generateName with label created-by-nab: `<nab-name>`
  - NAC lists VeleroBackup created by NAB by label as needed.
    - Since VeleroBackup name came from generateName (which apiServer already retries if collide with existing), we do not need a UUID logic inside NAC.

**Comment by shubham-pampattiwar** [comment]
yes this wasn't the issue, the problem would arise when we try to sync backups across clusters and in that case we wanted to avoid collisions, thats why we ended up using custom UUID.

**Comment by kaovilai** [comment]
If NAB syncs across cluster, wouldn't the generateName logic work in both clusters if both are synced to have the same NABs in the cluster?

**Comment by kaovilai** [comment]
Anyway yes as mpyrc said prob more important issues.. I just get excited at deleting code.

**Comment by kaovilai** [comment]
If NABs do not sync across cluster, it would be interesting to see how one cluster's NAC read the other.
